sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1368,6 +1368,17 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
     })
   }
 
+  test("SPARK-20144: Keep partition order if allowReordering=false.") {
+    withSQLConf(
+      "spark.sql.files.allowReordering" -> "false"
```

Comment:
If you change this to `true` (or revert the change) the test fails because the partitions come back in random order. E.g.:

```
Array([5], [6], [7], [8], [9], [0], [1], [2], [3], [4]) did not equal Array([0], [1], [2], [3], [4], [5], [6], [7], [8], [9]) (CSVSuite.scala:1378)
```